### PR TITLE
feat: Add GET paths for parents and children + fix: Update Neo4J query

### LINKED ISSUE #33: Add GET path for finding relationships of a node
There is no path for finding relationships (mainly `is_child_of` relationship) for any node. This must be added to compute the immediate parents and children of a particular node. (Will be helpful for the frontend)

### LINKED ISSUE #40: Update Neo4J query to disallow partial updates
Currently, the frontend tracks all changes (creation, addition, deletion) on keys and values of a node, and POSTs all information (such as translations, preceding_lines, properties etc) about a node. 

Since, the query only allows partial updates, deleted keys are not tracked in the database. Hence, the update db query has to be changed to allow full updates to a node.

### PR DESCRIPTION
**Pull request by aadarsh-ram**
### What
- Adds GET paths `/entry/{entry}/parents` and `/entry/{entry}/children` 
- Used for finding parents and children of a node
- Changed neo4j query to disallow partial updates

### Related issue(s)
- Fixes #33 
- Fixes #40

### CODE REVIEW

File: backend/editor/entries.py [hunk 1]
```diff
@@ -53,7 +75,7 @@ def update_nodes(label, entry, incomingData):
             raise ValueError("Invalid key: %s", key)
     
     # Build query
-    query = [f"""MATCH (n:{label}) WHERE n.id = $id"""]
+    query = [f"""MATCH (n:{label}) WHERE n.id = $id """, """SET n={} """, """SET n.id = $id"""]
```
Comment (alexgarel) [inline backend/editor/entries.py:78]: The idea here is that your reset the node before running updates ?

**Review by alexgarel — CHANGES_REQUESTED** [review]
It's ok but I have one question:

> * Changed neo4j query to disallow partial updates
Why do we want this ? Shan't it be a parameter or a specific url ?

Also if we have to push every key / values each time, we should refactor the code, because it can be done in only one query instead of so many.

File: backend/editor/entries.py (hunk 1 above)
Comment (aadarsh-ram) [reply backend/editor/entries.py:78]: Yess. The React App sends all the current keys of a node and the backend updates only those keys. But, if a user deletes any key, the change doesn't reflect. Hence, I clear the node and start adding all the keys sent by the React App.

Comment (aadarsh-ram) [reply backend/editor/entries.py:78]: If there's a better way to approach this, do let me know.

Comment (alexgarel) [reply backend/editor/entries.py:78]: Hi @aadarsh-ram, the way to remove a property could be to just set it to `null`

So if I want to remove the "tags_fr" key, I send "tags_fr: null".

**Comment by aadarsh-ram** [comment]
@alexgarel I have now figured out a new way to tackle this. I fetch the current node information and compare it with the node object received, to remove the deleted keys. Does this make sense?

File: backend/editor/entries.py [hunk 2]
```diff
@@ -74,9 +74,20 @@ def update_nodes(label, entry, incomingData):
         if not re.match(r"^\w+$", key) or key == "id":
             raise ValueError("Invalid key: %s", key)
     
+    # Get current node information and deleted keys
+    curr_node = list(get_nodes(label, entry).data()[0]['n'].keys())
+    deleted_keys = list(set(curr_node) ^ set(incomingData))
```
Comment (alexgarel) [inline backend/editor/entries.py:79]: Don't you want to just remove keys where incoming data is null ?

**Review by alexgarel — APPROVED** [review]
I made a comment, but this can be ok, also like this. So fix it if you want.

**Pull request merged**